packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -73,6 +74,34 @@ export interface ContractExchangeRates {
   [address: string]: number | undefined;
 }
 
+type MarketDataDetails = {
+  tokenAddress: `0x${string}`;
+  value: number;
+  currency: string;
+  allTimeHigh: number;
+  allTimeLow: number;
+  circulatingSupply: number;
+  dilutedMarketCap: number;
+  high1d: number;
+  low1d: number;
+  marketCap: number;
+  marketCapPercentChange1d: number;
+  price: number;
+  priceChange1d: number;
+  pricePercentChange1d: number;
+  pricePercentChange1h: number;
+  pricePercentChange1y: number;
+  pricePercentChange7d: number;
+  pricePercentChange14d: number;
+  pricePercentChange30d: number;
+  pricePercentChange200d: number;
+  totalVolume: number;
+};
+
+export type ContractMarketData = {
```

Comment:
We have a type called "market data" that has a property called "market data". This seems a bit misrepresentative.

Besides this, I guess I'm curious why this type is needed since it only has one property? It seems like in all the places where we return this object we could return a `Record<Hex, MarketDataDetails>` instead.

If that's the case, what if we rename `MarketDataDetails` to `TokenMarketData`? Then perhaps we don't need this type anymore — anywhere we need it, we can say `Record<Hex, TokenMarketData>`, and we can understand it to mean a collection of `TokenMarketData`s without having to give it a name.

What do you think?